### PR TITLE
[KYUUBI #6534] Fix checkEngineConnectionAlive may throw NPE

### LINKED ISSUE #6534: [Bug] checkEngineConnectionAlive may throw NPE
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

NPE occurs on launching Spark engine.

### Affects Version(s)

master/branch-1.9

### Kyuubi Server Log Output

```logtalk
java.lang.NullPointerException: Cannot invoke "org.apache.kyuubi.client.KyuubiSyncThriftClient.remoteEngineBroken()" because the return value of "org.apache.kyuubi.session.KyuubiSessionImpl.client()" is null
        at org.apache.kyuubi.session.KyuubiSessionImpl.checkEngineConnectionAlive(KyuubiSessionImpl.scala:321)
        at org.apache.kyuubi.session.KyuubiSessionManager.$anonfun$startEngineAliveChecker$2(KyuubiSessionManager.scala:421)
        at org.apache.kyuubi.session.KyuubiSessionManager.$anonfun$startEngineAliveChecker$2$adapted(KyuubiSessionManager.scala:418)
        at scala.collection.Iterator.foreach(Iterator.scala:943)
        at scala.collection.Iterator.foreach$(Iterator.scala:943)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
        at scala.collection.IterableLike.foreach(IterableLike.scala:74)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
        at org.apache.kyuubi.session.KyuubiSessionManager.$anonfun$startEngineAliveChecker$1(KyuubiSessionManager.scala:418)
        at org.apache.kyuubi.util.ThreadUtils$.$anonfun$scheduleTolerableRunnableWithFixedDelay$1(ThreadUtils.scala:126)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description
## Issue References 🔗

This pull request fixes #6534

## Describe Your Solution 🔧

check null ahead.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Pass GHA

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
@cfmcgrady this one is expected to be included in 1.9.2

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 1]
```diff
@@ -317,7 +317,8 @@ class KyuubiSessionImpl(
   }
 
   def checkEngineConnectionAlive(): Boolean = {
-    if (Option(client).exists(_.engineConnectionClosed)) return false
+    if (client == null) return true // client has not been initialized
```
Comment (cfmcgrady) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:320]: I'm unclear on the intention of this expression compared to its previous version.

when client is null, `Option(client).exists(_.engineConnectionClosed)` should be false?

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:320]: > when client is null, `Option(client).exists(_.engineConnectionClosed)` should be false?

correct, then it goes to the next line, and NPE occurs

**Review by cfmcgrady — APPROVED** [review]

**Review by turboFei — APPROVED** [review]
thanks, LGTM

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.9

**Pull request closed without merging**